crates/cheatnet/src/forking/state.rs

```diff
@@ -45,6 +52,58 @@ impl ForkStateReader {
     }
 }
 
+impl BlockInfoReader for ForkStateReader {
+    fn get_block_info(&mut self) -> StateResult<CheatnetBlockInfo> {
+        if let Some(cache_hit) = self.cache.get_block_info() {
+            return Ok(cache_hit);
+        }
+
+        match self
+            .runtime
+            .block_on(self.client.get_block_with_tx_hashes(self.block_id))
+        {
+            Ok(MaybePendingBlockWithTxHashes::Block(block)) => {
+                let block_info = CheatnetBlockInfo {
+                    block_number: BlockNumber(block.block_number),
+                    timestamp: BlockTimestamp(block.timestamp),
+                    sequencer_address: ContractAddress(patricia_key!(block.sequencer_address)),
+                };
+
+                self.cache.cache_get_block_info(block_info);
+
+                Ok(block_info)
+            }
+            Ok(MaybePendingBlockWithTxHashes::PendingBlock(pending_block)) => {
```

Comment:
This is barely comprehensible, can we extract getting the block info into a method?

Reply:
updated